src/components/HeatMap/index.tsx

```diff
@@ -0,0 +1,282 @@
+import { useNavigation } from "@react-navigation/native";
+import React, { useEffect, useState } from "react";
+import { Polygon } from "react-native-maps";
+import { useTheme } from "styled-components";
+
+import { StayNormalMap } from "../../screens/Home/styles";
+import { getCoordinates } from "../../services/coordinateSecretary";
+import staySafeDarkMapStyle from "../../styles/staySafeDarkMapStyle";
+import CircularLoader from "../CircularLoader";
+import { coordinates } from "./coordinates";
+
+interface CoordinateCitiesDF {
+    name: string;
+    color: string;
+    coordinates: Array<Coordinate>;
+}
+
+interface Coordinate {
+    longitude: number;
+    latitude: number;
+}
+
+interface HeatMapProps {
+    secretaryOccurrences: Array<CityCrimes>;
+    city: string;
+}
+
+interface CityCrimes {
+    name: string;
+    crimes: Array<Crime>;
+}
+
+interface Crime {
+    nature: string;
+    quantity: number;
+    classification: number;
+}
+
+interface CitiesCrimes {
+    name: string;
+    nature: string;
+    quantity: number;
+    classification: number;
+    color?: string;
+}
+
+interface CoordinateResponse {
+    cities: Array<CityCoordinate>;
+}
+
+interface CityCoordinate {
+    name: string;
+    coordinates: Array<Array<number>>;
+}
+
+const HeatMap: React.FC<HeatMapProps> = ({ secretaryOccurrences, city }) => {
+    const navigation = useNavigation();
+    const theme = useTheme();
+
+    const [citiesCrimes, setCitiesCrimes] = useState<CitiesCrimes[]>([]);
+
+    const [isSelected, setIsSelected] = useState(false);
+    const [selectedRegion, setSelectedRegion] = useState("");
+
+    const [coordinatesSp, setCoordinatesSp] = useState([]);
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        setIsLoading(true);
+        if (city === "df") {
+            setCrimes();
+            setIsLoading(false);
+        } else {
+            setCrimes();
+            getSpCoordinates().then((res) => setIsLoading(false));
+        }
+    }, []);
+
+    const getSpCoordinates = async () => {
+        const response = await getCoordinates(city);
+
+        setCoordinatesSp(
```

Comment:
if the function is just a get, it should return the coordinates, or you could change the funcion name